server/src/main/java/org/elasticsearch/index/mapper/AbstractGeometryFieldMapper.java

```diff
@@ -399,6 +425,9 @@ public void doXContentBody(XContentBuilder builder, boolean includeDefaults, Par
         if (includeDefaults || ignoreZValue.explicit()) {
             builder.field(GeoPointFieldMapper.Names.IGNORE_Z_VALUE.getPreferredName(), ignoreZValue.value());
         }
+        if (includeDefaults || docValues.explicit()) {
```

Comment:
Hi @jpountz, in a follow-up commit I've removed the Explicit docValues field from the Mapper, but without it, this conditional is difficult to achieve (unless I'm missing something). It is difficult to check whether the value was set explicitly without access to `docValuesSet` which is in the field mapper's builder.

not sure this is a big deal, but it does feel like the doc values field should only be serialized in this scenaior